pkg/datapath/linux/probes/probes.go

```diff
@@ -151,6 +151,14 @@ type Features struct {
 	Misc         `json:"misc"`
 }
 
+// Prober abstracts the notion of a kernel feature prober. This is useful for
+// testing purposes as it allows us to mock out the kernel, enabling control
+// control over what features are returned.
+type Prober interface {
+	// Probe returns the kernel feaures available on machine.
+	Probe() Features
+}
+
```

Comment:
Good point. I initially thought to provide the interface inside the `probes` package simply because proximity and that `pkg/bpf` was already importing `probes` anyway. However, that doesn't preclude the pattern of providing the interface at the consumer site, so I've made that change.